src/structure_dump.cpp

```diff
@@ -414,6 +414,55 @@ void structure::set_chiP_from_file(h5file *file, const char *dataset, field_type
   }
 }
 
+binary_partition::binary_partition(int _id) {
+  id = _id;
+  split_dir = NO_DIRECTION;
+  split_pos = 0.0;
+  left = NULL;
+  right = NULL;
+}
+
+binary_partition::binary_partition(direction _split_dir, double _split_pos) {
+  split_dir = _split_dir;
+  split_pos = _split_pos;
+  id = -1;
+  left = NULL;
+  right = NULL;
+}
+
+static void split_by_binarytree(grid_volume gvol,
+                                std::vector<grid_volume> &result_gvs,
+                                std::vector<int> &result_ids,
+                                const binary_partition *bp) {
```

Comment:
One option would be to change `split_by_const` etcetera so that they actually return a `binary_partition` and then we call `split_by_binarytree`.    That would make it easy to export the tree if we want, and would ensure that the `binary_partition` code gets further testing (because now all splitting would pass through it).